packages/manager/src/features/CloudPulse/Widget/CloudPulseWidget.tsx

```diff
@@ -255,7 +254,7 @@ export const CloudPulseWidget = (props: CloudPulseWidgetProperties) => {
     data = generatedData.dimensions;
     legendRows = generatedData.legendRowsData;
     today = generatedData.today;
-    currentUnit = generatedData.unit;
+    scaledWidgetUnit.current = scaledWidgetUnit.current !== generatedData.unit ? generatedData.unit : scaledWidgetUnit.current; // here state doesn't matter, as this is always the latest re-render
```

Comment:
```suggestion
    scaledWidgetUnit.current = generatedData.unit // here state doesn't matter, as this is always the latest re-render
```
condition not required as value will always be generatedData.unit